### PR TITLE
IGVF-145 display schemas and hierarchy

### PR DESCRIPTION
**Pull request by emro**
- Adds hierarchy view of schemas
- Displays schemas and changelogs

### CODE REVIEW

File: pages/profiles/index.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+          {Object.keys(props.child).map((child_key) => {
+            return (
+              <ChildElement
+                title={props.schemas[child_key] ? props.schemas[child_key].title : child_key}
+                schemaKey={child_key}
+                schemas={props.schemas}
+                child={props.child[child_key]}
+                indentation={(props.indentation + 4)}
+                key={child_key}
+              />
```
Comment (forresttanaka) [inline pages/profiles/index.js:34]: I always love recursive React components; I guess it looks cool (and of course it solves problems).

File: libs/breadcrumbs.js [hunk 2]
```diff
@@ -59,6 +81,9 @@ const buildBreadcrumbs = async (itemOrCollection, titleProp = "") => {
   } else if (itemOrCollection["@type"].includes("Item")) {
     // To handle the case in which `itemOrCollection` is an item...
     breadcrumbs = await buildItemBreadcrumbs(itemOrCollection, titleProp)
+  } else if (itemOrCollection["@type"].includes("JSONSchemas") || itemOrCollection["@type"].includes("JSONSchema")) {
+    // To handle the schema pages...
+    breadcrumbs = await buildSchemaBreadcrumbs(itemOrCollection, titleProp)
```
Comment (forresttanaka) [inline libs/breadcrumbs.js:86]: These lines are too long for prettier.

**Comment by forresttanaka** [comment]
One thing I forgot to mention, could you allow the schema pages to scroll horizontally? Some of them are _really_ wide. https://igvf-ui-igvf-145-profiles-page.demo.igvf.org/profiles/award

**Comment by forresttanaka** [comment]
The [demo](https://igvf-ui-igvf-60-report-view.demo.igvf.org/) has updated.

File: pages/profiles/[profile].js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    <>
+      <Breadcrumbs />
+      <PagePreamble />
+      <DataPanel>
+        <div className="font-semibold">Schema</div>
+        <div className="overflow-x-auto border border-gray-300 bg-gray-100 text-xs dark:border-gray-800 dark:bg-gray-900">
+          <pre className="p-1">{JSON.stringify(schema, null, 4)}</pre>
+        </div>
+      </DataPanel>
+      <br />
```
[the remark this answers is not in the document]
Comment (emro) [reply pages/profiles/[profile].js]: It's now using <DataTitle>

File: pages/profiles/[profile].js [hunk 4]
```diff
@@ -0,0 +1,57 @@
+// node_modules
+import PropTypes from "prop-types"
+import { marked } from 'marked'
```
[the remark this answers is not in the document]
Comment (emro) [reply pages/profiles/[profile].js]: I installed prettier and ran it on save

File: pages/profiles/index.js (hunk 1 above)
Comment (emro) [reply pages/profiles/index.js]: I actually couldn't think of another way to build the hierarchy!

**Review by forresttanaka — DISMISSED** [review]
I think it looks really good, and the horizontal scrolling works great. I added one optional suggestion for styling the changelogs. I had assumed the first time that the changelogs just had little styling, but they actually do have quite a lot.

File: pages/profiles/[profile].js [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      <DataPanel>
+        <div className="border border-gray-300 bg-gray-100 text-xs dark:border-gray-800 dark:bg-gray-900">
+          <pre className="p-1 overflow-x-scroll">{JSON.stringify(schema, null, 4)}</pre>
+        </div>
+      </DataPanel>
+      <DataAreaTitle>Changelog</DataAreaTitle>
+      <DataPanel>
+        <div className="border border-gray-300 bg-gray-100 text-xs dark:border-gray-800 dark:bg-gray-900">
+          <div dangerouslySetInnerHTML={{ __html: html }} />
+        </div>
```
[the remark this answers is not in the document]
Comment (forresttanaka) [reply pages/profiles/[profile].js:27]: I think the typography module probably styles the `<h1>` with a big top margin and maybe other things not so welcome, but they provide an easy mechanism to adjust their default styles.

[the remark this answers is not in the document]
Comment (emro) [reply pages/profiles/[profile].js:27]: This should definitely be fixed/improved and I'd like to address it in a follow-up ticket.

**Pull request merged**